librecad/src/ui/forms/qg_commandwidget.cpp

```diff
@@ -155,7 +157,48 @@ void QG_CommandWidget::tabPressed() {
             leCommand->setText(reducedChoice.first());
         }
         else if (reducedChoice.count()>0) {
+            QString longestString = "";
+            QString shortestString = "";
+            int lengthShortestString(0);
+            int lengthReducedChoice = reducedChoice.count();
+
+            // Finding which is the longest string
+            for(QStringList::Iterator it = reducedChoice.begin(); it != reducedChoice.end(); ++it) {
+                if((*it).length() > longestString.length()) {
+                    longestString = (*it);
+                }
+            }
+            int lengthLongestString = longestString.length();
+
+            // Finding which is the shortest string
+            lengthShortestString = longestString.length();
+            for(QStringList::Iterator it = reducedChoice.begin(); it != reducedChoice.end(); ++it) {
+                if((*it).length() < lengthShortestString) {
+                    shortestString = (*it);
+                }
+            }
+            lengthShortestString = shortestString.length();
+
+            // Now we parse the reducedChoice list, character of each item by character.
+            int i(0);
+            int pos(0);
+            bool common = true;
+
+            for(i = 0; i < lengthShortestString; ++i) {
+                for(QStringList::Iterator it = reducedChoice.begin(); it != reducedChoice.end(); ++it) {
+                    if(longestString.at(i) != (*it).at(i)) {
+                        common = false;
+                        break;
+                    }
+                }
+                if(common == true) {
+                    ++pos;
+                }
```

Comment:
yes, but foreach() is Qt preprocessor, while [range-based loop](http://en.cppreference.com/w/cpp/language/range-for) is c++11:

for(auto str: reducedChoice) is okay, because of QString default data sharing, but there's no drawback by specifying it's a const reference.